### PR TITLE
expose date in prow/cmd/deck jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes/test-infra/issues/3714

### CODE REVIEW

File: prow/cmd/deck/jobs.go [hunk 1]
```diff
@@ -189,7 +189,7 @@ func (ja *JobAgent) update() error {
 			ft: j.Status.CompletionTime,
 		}
 		if !nj.ft.IsZero() {
-			nj.Finished = nj.ft.Format("15:04:05")
+			nj.Finished = nj.ft.Format(time.Stamp)
```
Comment (BenTheElder) [inline prow/cmd/deck/jobs.go]: Actually, while `time.Stamp` mimics the `started` field, `time.RFC3339Nano` seems provides the most detailed information for programmatic use and I don't believe this is used anywhere else anyhow. I will change this.

**Comment by BenTheElder** [comment]
I didn't do start time because it *is* exposed on the web interface and
we'd have to parse it then.

**Comment by spxtr** [comment]
Right. If you want to do that in a follow-up PR I'd appreciate it. @rmmh has been bugging me for a while to use local times for that...

**Comment by rmmh** [comment]
Let's just do RFC3339. Nanosecond precision is overkill here. :-)

We could also export the seconds directly, then in the browser render with something like `new Date(build.started * 1000).toLocaleString()` => "7/13/2017, 7:40:00 PM"

**Comment by rmmh** [comment]
You also need to bump the deck version before deploying this.

**Pull request merged**